elements/ip6/storeip6address.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * storeip6address.{cc,hh} -- element stores IPv6 destination annotation into
+ * packet
+ * Glenn Minne
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, subject to the conditions
+ * listed in the Click LICENSE file. These conditions include: you must
+ * preserve this copyright notice, and you cannot mention the copyright
+ * holders in advertising related to the Software without their permission.
+ * The Software is provided WITHOUT ANY WARRANTY, EXPRESS OR IMPLIED. This
+ * notice is a summary of the Click LICENSE file; the license in that file is
+ * legally binding.
+ *
+ * For using the 'src' and 'dst argument the IPv6 header pointers MUST be set.
+ */
+
+#include <click/config.h>
+#include "storeip6address.hh"
+#include <click/args.hh>
+#include <click/error.hh>
+#include <clicknet/ip6.h>
+#include <clicknet/tcp.h>
+#include <clicknet/udp.h>
+CLICK_DECLS
+
+StoreIP6Address::StoreIP6Address()
+{
+}
+
+StoreIP6Address::~StoreIP6Address()
+{
+}
+
+int
+StoreIP6Address::configure(Vector<String> &conf, ErrorHandler *errh)
+{
+
+    String offset;	// the offset to be read; can be equal to src or dst in stead of a number, in which case this has a special meaning.
+    int r;		// reply
+
+    if (conf.size() == 1) {
+	      r = Args(conf, this, errh).read_mp("OFFSET", WordArg(), offset).complete();
+        _address_given = false;
+    } else if (conf.size() == 2) {
+	      r = Args(conf, this, errh).read_mp("ADDR", _address).read_mp("OFFSET", WordArg(), offset).complete();
```

Comment:
Prefer an approcach parsing the two, and then if ADDRESS is unset, do something. Does that seem possible?